tests/Sinch.Tests/e2e/Conversation/TemplatesTests.cs

```diff
@@ -0,0 +1,96 @@
+
+        private readonly Template _template = new Template()
+        {
+            Id = "template_id",
+            Description = "template",
+            CreateTime = DefaultTime,
+            DefaultTranslation = "en-US",
+            UpdateTime = DefaultTime,
+            Version = 3,
+            Translations = new List<TemplateTranslation>()
+            {
+                new TemplateTranslation(new TextMessage("hello ${a}"))
+                {
+                    Version = "1", // is this really a string?
+                    CreateTime = DefaultTime,
+                    UpdateTime = DefaultTime,
+                    LanguageCode = "en-US",
+                    ChannelTemplateOverrides = null,
+                    Variables = new List<TypeTemplateVariable>()
+                    {
+                        new TypeTemplateVariable()
+                        {
+                            Key = "a",
+                            PreviewValue = "c"
+                        }
+                    }
+                }
+            },
+        };
+
+        [Fact]
+        public async Task Get()
+        {
+            var response = await SinchClientMockServer.Conversation.TemplatesV2.Get("template_id");
+            response.Should().BeEquivalentTo(_template);
+        }
+
+        [Fact]
+        public async Task Delete()
+        {
+            var op = () => SinchClientMockServer.Conversation.TemplatesV2.Delete(_template.Id);
+            await op.Should().NotThrowAsync();
+        }
+
+        [Fact]
+        public async Task List()
+        {
+            var response = await SinchClientMockServer.Conversation.TemplatesV2.List();
+            response.Should().BeEquivalentTo(new List<Template>() { _template, _template });
+        }
+
+        [Fact]
+        public async Task Create()
+        {
+            var createTemplateRequest = new CreateTemplateRequest()
+            {
+                Description = _template.Description,
+                Translations = _template.Translations,
+                CreateTime = _template.CreateTime,
+                DefaultTranslation = _template.DefaultTranslation,
+                UpdateTime = _template.UpdateTime,
+                Version = _template.Version
+            };
+            var response = await SinchClientMockServer.Conversation.TemplatesV2.Create(createTemplateRequest);
+            response.Should().BeEquivalentTo(_template);
+        }
+
+        [Fact]
+        public async Task ListTranslations()
+        {
+            var response =
+                await SinchClientMockServer.Conversation.TemplatesV2.ListTranslations(_template.Id, "en-US", "3");
+            response.Should().BeEquivalentTo(_template.Translations);
+        }
+
+        [Fact]
+        public async Task Update()
+        {
+            var response = await SinchClientMockServer.Conversation.TemplatesV2.Update(_template);
```

Comment:
Test to be updated according to how you treat the previous comment regarding the `id` property